log/formatter.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package log
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+//go:generate mockery --name DirectoryContentFormatter --case=underscore
+
+// DirectoryContentFormatter allows customization of the directory output
+type DirectoryContentFormatter interface {
```

Comment:
This is an interesting API. Could we also add some examples of how these fields affect the logged output? I'm having a hard time picturing how RootPath for example will be used by an end user without them having to dive into logging internals.